Source/Extensions/Blazorise.TreeView/Extensions/NodeStateExtensions.cs

```diff
@@ -24,5 +25,23 @@ internal static class NodeStateExtensions
                 yield return new TreeViewNodeState<TNode>( node, hasChildren, expanded( node ) );
             }
         }
+
+        public static async IAsyncEnumerable<TreeViewNodeState<TNode>> ToNodeStates<TNode>( this IList nodes,
```

Comment:
Shouldn't just check / cast the IList to an IList of TNode and call the `IEnumerable` version?

Reply:
It's not a generic type

![image](https://user-images.githubusercontent.com/900302/229787232-5b2c8924-f023-43a8-86f0-a676fed0b455.png)

Reply:
That's strange, what's the original type the debugger tells you? O.o I'm pretty sure you should be able to get it to work with a generic collection.

You also have stuff like this:
https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.cast?view=net-8.0

Anyway it isn't a big deal otherwise both methods are easily viewable in the same file, doubt we'd make the mistake to only update one in the future if any refactor was needed.

Reply:
![image](https://user-images.githubusercontent.com/900302/229790608-e3b43650-f201-4b38-a82a-05df184cd355.png)

Reply:
Ahh, it stems from the collection notifier already casting down. Damn those APIs from Windows forms eras... :D 

So in my opinion the change should be something like 
`e.NewItems.Cast<TNode>().ToNodeStates`

or `OfType`